eventrouter/internal/queue_posix.c

```diff
@@ -0,0 +1,237 @@
+#include "queue_.h"
```

Comment:
Why did you abandon the idea of using message queues or sockets (your initial suggestion)? 

Seems like that would have been a lot simpler.

Reply:
> I abandoned the `mqueue` approach because it isn't supported on OSX and may cause contention due to their manifestation in the file tree.

OSX never fails to disappoint....

> 
> I didn't use sockets because I confused them with FIFOs and thought managing two ends of the pipe would be painful. I'm unsure why I didn't revisit the thought while implementing this; I think I started having fun.
> 
> If we can do the same with sockets, which we probably can, I should probably replace this implementation with that.

I don't see why it wouldn't work. Both `read()` and `write()` are required by POSIX to be thread-safe (well, atomic), we only read and write fixed-size chunks of data (`sizeof(void*)`?), so it should be OK.

Reply:
If possible, I would like to leave the sockets-based implementation for a subsequent PR. I'm having trouble getting it working.

Reply:
As I commented before, I think this PR is fine as it is. Unblocking further work is more important than ironing out the implementation details.